src/geode/geosciences/utils.cpp

```diff
@@ -89,5 +91,46 @@ namespace geode
             };
             return "";
         }
+
+        absl::optional< std::string > goto_keyword_if_it_exists(
+            std::ifstream& file, absl::string_view word )
+        {
+            std::string line;
+            while( std::getline( file, line ) )
+            {
+                if( string_starts_with( line, word ) )
+                {
+                    return line;
+                }
+            }
+            geode::Logger::debug(
+                "[goto_keyword_if_it_exists] Couldn't find word ", word,
+                " in the file, returning to file begin." );
+            file.clear();
+            file.seekg( 0 );
```

Comment:
Should it be 0, or the same line it was at the beginning of the method?

Reply:
I'd say both can be an answer, but anyways I didn't find a way to save the line index, to go back to it here.

Reply:
https://en.cppreference.com/w/cpp/io/basic_istream/tellg